### PR TITLE
feat: Add `--relay=<url>/disable/default` options and `--ticket_type` option for send

### PR DESCRIPTION
**Pull request by matheus23**
I was using these options to test stuff - I think they could be really useful to diagnose issues from the CLI.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -107,7 +111,18 @@ pub struct CommonArgs {
 
     #[clap(short = 'v', long, action = clap::ArgAction::Count)]
     pub verbose: u8,
+
+    /// The relay URL to use as a home relay.
+    ///
+    /// If not set, sendme will use the default relay set from number 0.
+    #[clap(long)]
+    pub relay: Option<String>,
```
Comment (rklaehn) [inline src/main.rs]: Can we make this an enum?

enum Relay {
  Auto,
  Url(Url),
  Off,
}

or something?

--relay auto (default)
--relay off
--relay url(https://bla) or even just https://

?

File: src/main.rs [hunk 2]
```diff
@@ -116,6 +131,13 @@ pub struct SendArgs {
     /// being shared.
     pub path: PathBuf,
 
+    /// Whether to use a short ticket that doesn't contain information
+    /// about your IP addresses.
+    ///
+    /// Useful if you want smaller tickets or want to test node discovery mechanisms.
+    #[clap(long, action = clap::ArgAction::SetTrue)]
+    pub short_ticket: bool,
```
Comment (rklaehn) [inline src/main.rs]: Can we just use https://docs.rs/iroh/latest/iroh/base/node_addr/enum.AddrInfoOptions.html here? It seems to support FromStr, so it should just work here.

**Comment by rklaehn** [comment]
Agree that this is useful. Added a few comments about how the cli args should look.

File: src/main.rs (hunk 1 above)
Comment (matheus23) [reply src/main.rs]: I've made it similar to the terminology in the code. "disabled" and "default".

**Review by rklaehn — APPROVED** [review]

**Comment by rklaehn** [comment]
Wait a minute... Should we fold in https://github.com/n0-computer/sendme/pull/32 ?

**Comment by matheus23** [comment]
Ah - just saw your comment. I've created #38 which interestingly works different from #32.

I guess #32 makes more sense :thinking:

**Comment by matheus23** [comment]
Alright. I took the approach from #32 and incorporated it in #38 :+1:

**Comment by rklaehn** [comment]
> Ah - just saw your comment. I've created #38 which interestingly works different from #32.
> 
> I guess #32 makes more sense 🤔

Yes, sendme has clearly defined roles. The sender does not need to resolve nodes, only publish its own node id. And the receiver does not need to publish, only resolve.

So I think there is no harm in enabling the resolve on the receive side always, but on the send side you really want to only publish if you are in the right mode, and have some comment in the help about this.

Let's imagine you are a whistleblower and want to use sendme to send some data to a publisher. You **don't** want your node id, public ip address, and some information about your local network setup to be blasted on dns.iroh.link.

Sendme by default should be just between the sender and the receiver, with just a tiny bit of unavoidable interaction with the relay. Same with dumbpipe, btw.

**Comment by rklaehn** [comment]
> > Ah - just saw your comment. I've created #38 which interestingly works different from #32.
> > I guess #32 makes more sense 🤔
> 
> Yes, sendme has clearly defined roles. The sender does not need to resolve nodes, only publish its own node id. And the receiver does not need to publish, only resolve.
> 
> So I think there is no harm in enabling the resolve on the receive side always, but on the send side you really want to only publish if you are in the right mode, and have some comment in the help about this.
> 
> Let's imagine you are a whistleblower and want to use sendme to send some data to a publisher. You **don't** want your node id, public ip address, and some information about your local network setup to be blasted on dns.iroh.link.
> 
> Sendme by default should be just between the sender and the receiver, with just a tiny bit of unavoidable interaction with the relay. Same with dumbpipe, btw.

**Pull request merged**